src/cmd/top/summary_pane.c

```diff
@@ -429,6 +447,31 @@ static void stats_continuation (flux_future_t *f, void *arg)
             fatal (errno, "error getting job-list.job-stats RPC response");
     }
 
+    if (sum->top->queue) {
+        json_t *queues = NULL;
+        json_t *q = NULL;
+        size_t index;
+        json_t *value;
+        if (json_unpack (o, "{s:o}", "queues", &queues) < 0)
+            fatal (0, "error getting job-list.job-stats queues");
+        if (json_is_array (queues) == 0)
+            fatal (EPROTO, "job-list.job-stats queue not an array");
+        json_array_foreach (queues, index, value) {
+            const char *name;
+            if (json_unpack (value, "{s:s}", "name", &name) < 0)
+                fatal (EPROTO, "job-list.job-stats queue name not found");
+            if (!strcmp (name, sum->top->queue)) {
+                q = value;
+                break;
+            }
+        }
```

Comment:
Suggestion: move his  block to its own function, something like
```c
json_t *parse_queue_stats (json_t *o, const char *name)
```
and have it return NULL on failure, then handle the failure in one place as a protocol error rather than having three separate error message for what is basically the same unlikely problem.